File: Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        // Note: The Solution object is instantiated only once and is reused by each test case.
        int index = 1, count = 1, len = 1;
        if(!n){
            return 0;
        }
        for(int i=1; i<n; i++){
            if(A[i]!=A[i-1]){
                count = 1;
                A[index++] = A[i];
                len++;
            }
            else if(count < 2){
                A[index++] = A[i];
                count++;
                len++;
            }
            else{
                count++;
            }
        }
        return len;
    }
};

File: java/Remove Duplicates from Sorted Array II.java
public class Solution {
    public int removeDuplicates(int[] A) {
        int len = A.length;
        int index = 0;
        if(len <= 2) {
            return len;
        }
        index = 1;
        for(int i=2; i<len; i++) {
            if(A[i] == A[index] && A[i] == A[index-1]) {
                continue;
            }
            else {
                A[++index] = A[i];
            }
        }
        return index+1;
    }
}